pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -180,17 +180,24 @@ func (m *managerImpl) Start() error {
 	if err != nil {
 		return err
 	}
+	timer := time.NewTimer(dbusReconnectPeriod)
+
 	go func() {
+		defer timer.Stop()
 		for {
-			if stop != nil {
-				<-stop
-			}
+			select {
+			case <-timer.C:
```

Comment:
1. do we really need a `select` block here? since there is only one case, maybe directly use `<- timer.C` is more simplier.
2. using `wait.Forever` seems much more clear and simpler here. the code will looks like:
```go
go func () {
    if stop != nil {
       <-stop
    }
    wait.Forever(func(){
        m.logger.V(1).Info("Restarting watch for node shutdown events")
        stop, err := m.start()
        if err != nil {
           m.logger.Error(err, "Unable to watch the node for shutdown events")
           return
        }
        <- stop
    }, dbusReconnectPeriod)
}
```

Reply:
Good suggestion, thanks. 
It is more appropriate to use internal packages

Reply:
After discussing offline with @xuzhenglun , it seems that using `time.Timer` is more clearer.